tools/js2c.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#  This file converts ./js/*.js to a C-array in ./source/jerry-targetjs.h file
+
+import argparse
+import glob
+import os
+import re
+import c_source_helper
+
+HEADER = '''#ifndef JERRY_TARGETJS_H
+#define JERRY_TARGETJS_H
+'''
+
+FOOTER = '''
+#endif
+'''
+
+NATIVE_STRUCT = '''
+struct js_source_all {
+  const char* name;
+  const char* source;
+  const int length;
+};
+
+#define DECLARE_JS_CODES \\
+struct js_source_all js_codes[] = \\
+{ \\'''
+
+
+def extract_name(path):
+    special_chars = re.compile(r'[-\\?\'".]')
+    return special_chars.sub('_', os.path.splitext(os.path.basename(path))[0])
+
+
+def reduce_code(code):
+    code = re.sub(re.compile(r"/\*.*?\*/", re.DOTALL), "", code)  # remove all occurance streamed comments
+    code = re.sub(re.compile(r"//.*?\n"), "", code)  # remove all occurance singleline comments
+    code = re.sub('\n+', '\n', re.sub('\n +', '\n', code))  # remove white spaces
+    return code
+
+
+def js_to_native_code(path, name, build_type):
+    code = {}
```

Comment:
I don't think we need this line here. The `code` will be overwritten in line 61.